pkg/component/certmanagement/certmanagement.go

```diff
@@ -0,0 +1,145 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package certmanagement
+
+import (
+	"context"
+	"fmt"
+
+	certv1alpha1 "github.com/gardener/cert-management/pkg/apis/cert/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/component"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+)
+
+const (
+	// componentName is the name of the cert-management component.
+	componentName = "cert-management"
+	// issuersManagedResourceName is the name of the issuers ManagedResource.
+	issuersManagedResourceName = "cert-management-issuers"
+	// DefaultIssuerName is the name of the default issuer.
+	DefaultIssuerName = "default-issuer"
+
+	appName     = "app.kubernetes.io/name"
+	appInstance = "app.kubernetes.io/instance"
+)
+
+type certManagement struct {
+	values     Values
+	client     client.Client
+	deployment component.DeployWaiter
+}
+
+// Values is a set of configuration values for the cert-management component.
+type Values struct {
+	Image         string
+	Namespace     string
+	Deployment    *operatorv1alpha1.CertManagementDeployment
+	DefaultIssuer operatorv1alpha1.DefaultIssuer
+}
+
+var listOpts = []client.ListOption{
+	client.InNamespace(v1beta1constants.GardenNamespace),
+	client.MatchingLabels{"app.kubernetes.io/name": componentName},
+}
+
+// New creates a new Deployer for the cert-management component.
+func New(
+	cl client.Client,
+	values Values,
+) component.DeployWaiter {
+	return &certManagement{
+		values:     values,
+		client:     cl,
+		deployment: newCertManagementDeployment(cl, values),
+	}
+}
+
+var _ component.DeployWaiter = &certManagement{}
+
+func (c *certManagement) Deploy(ctx context.Context) error {
+	if err := c.deployment.Deploy(ctx); err != nil {
+		return err
+	}
+	return c.deployDefaultIssuer(ctx)
```

Comment:
We usually create one managed resource per component. Could you split the cert-management and default-issuer components?
There are similar cases like [prometheus-operator](https://github.com/gardener/gardener/tree/master/pkg/component/observability/monitoring) and [fluent-operator](https://github.com/gardener/gardener/tree/master/pkg/component/observability/logging) which are done this way

Reply:
refactored